### ISSUE TITLE
Clarification needed on baseline reproduction details for Llama-3.2-1B-IT on GSM8K

### ISSUE THREAD

**Issue by james-yw** [bryanchrist/MathNeuro#5]
Hi, thanks for sharing your excellent work on MathNeuro!

When reproducing the baseline results of **Llama-3.2-1B-IT** on **GSM8K**, I used the setting `random_state=42` and `eval_dataset_subset=200`. Under these conditions, I obtained an exact match accuracy of about **0.365** on GSM8K:

```json
{
  "gsm8k_cot": {
    "alias": "gsm8k_cot",
    "exact_match,strict-match": 0.365,
    "exact_match_stderr,strict-match": 0.0341,
    "exact_match,flexible-extract": 0.365,
    "exact_match_stderr,flexible-extract": 0.0341
  }
}
```

However, in Figure 51 (page 33) of the paper, the reported baseline accuracy for Llama-3.2-1B-IT on GSM8K is **0.34**.

Could you please provide the exact `random_state`, `eval_dataset_subset`, and any other relevant experimental settings used during evaluation? This will help to reproduce the results accurately.

Without this information, I am concerned that the reported baseline might be underestimated, which could inadvertently exaggerate the improvement brought by MathNeuro. Accurate baselines are crucial for fair comparisons.

Thank you!

**Comment by bryanchrist** [comment]
We agree accurate baselines are crucial, which is why we ran the baseline several times for each model to ensure we consistently observed the same results. We also re-ran the Llama 1B baseline following your comment, and still got the 34% accuracy reported in the paper. Running the code snippet for the pruning experiments below should yield the same results in a pre_train json object. If not, it is possible that different hardware could be causing the discrepancy. 

```
python MathNeuro.py --model meta-llama/Llama-3.2-1B-Instruct --save_path /results_path --train_dataset data/gsm8k.csv --eval_datasets race mmlu --calibration_datasets data/race.csv data/mmlu.csv --eval_dataset_subset 200 --calibration_dataset_names Race MMLU --train_lm_eval_task gsm8k_cot --pre_train_eval
```
